cluster/gce/gci/configure-kubeapiserver.sh

```diff
@@ -101,9 +101,6 @@ function start-kube-apiserver {
     params+=" --service-account-key-file=${SERVICEACCOUNT_CERT_PATH}"
   fi
   params+=" --token-auth-file=/etc/srv/kubernetes/known_tokens.csv"
-  if [[ -n "${KUBE_PASSWORD:-}" && -n "${KUBE_USER:-}" ]]; then
-    params+=" --basic-auth-file=/etc/srv/kubernetes/basic_auth.csv"
```

Comment:
there's a lot of plumbing to create this file and manage KUBE_PASSWORD... not sure how much should be cleaned up in this PR or not (e.g. in get-kubeconfig-basicauth, create-kubeconfig, get-kubeconfig-user-basicauth, gen-kube-basicauth, load-or-gen-kube-basicauth, create-master-auth, etc)

Reply:
@bentheelder @mikedanese @spiffxp @dims you guys are listed as approvers for the `cluster` dir - do you have an opinion on this?  I try to stay as far away from BASH as possible, but if y'all think it is worthwhile to clean it up, I can look into it.

Reply:
@cheftako for GCI changes.

I would keep to minimum changes in cluster/ generally, if we decide to clean this up at some point it'll be a deep scrub 🙃

Reply:
On reflection, I agree